src/test/java/page/GoogleHomePage.java

```diff
@@ -0,0 +1,36 @@
+package page;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+public class GoogleHomePage extends BasePage {
+
+    private static final String GOOGLE_HOMEPAGE_URL = "https://www.google.com/";
+    private WebDriver driver;
+
+    @FindBy(name = "q")
+    private WebElement googleSearchField;
+
+    @FindBy(name = "btnK")
+    private WebElement googleSearchButton;
+
+    public GoogleHomePage(WebDriver driver) {
+        this.driver = driver;
+
+        PageFactory.initElements(driver, this);
+    }
+
+    public GoogleHomePage openGooglePage() {
+        pageOpener(GOOGLE_HOMEPAGE_URL);
+        return this;
+    }
+
+    public GoogleSearchForWikipediaResultsPage searchForTerms(String searchTerm) {
+        pageOpener(GOOGLE_HOMEPAGE_URL);
+        googleSearchField.sendKeys(searchTerm);
+        googleSearchButton.submit();
+        return new GoogleSearchForWikipediaResultsPage(driver, searchTerm);
```

Comment:
why you need searchTerm in you return statement?

Reply:
I'm using it for the search field actually. It wouldn't work otherwise.